src/Events/PatientDocuments/PatientRetrieveOffsiteDocument.php

```diff
@@ -18,19 +18,24 @@ class PatientRetrieveOffsiteDocument extends Event
 {
     const REMOTE_DOCUMENT_LOCATION = 'remote.document.retrieve.location';
     private string $url;
-    private $offsiteurl;
-    public function __construct($url)
+    private $offsiteUrl;
+    public function __construct(string $url)
     {
         $this->url = $url;
     }
 
-    public function setOffsiteUrl(string $offsitedUrl): void
+    public function getOpenEMRDocumentUrl(): string
```

Comment:
@juggernautsei Please look over this file.  I exposed the $url property via getOpenEMRDocumentUrl() and fixed the setter in the setOffsiteUrl.

I'm not sure how this was working for you before, but it should be working now.

Reply:
@adunsulag I see the temp_url being passed in whole. The getOpenEMRDocumentUrl() may not be needed for me. But it could be useful for the future of someone else's module. My module catches the $this->url already. Then I will move the logic into there to get the file name.  

Now my logic in the module looks like this.

    public function retrieveFromS3(PatientRetrieveOffsiteDocument $event): void
    {
        $url = $event->url;
        $s3Key = explode("//", $url); //split the url to get the s3 key
        $s3model = new S3Model();
        $credentials = new Credentials($s3model);
        $s3Client = $credentials->authenticateCredentials();
        $getBucket = $s3model->getAllCredentials();
        $cmd = $s3Client->getCommand('GetObject', [
            'Bucket' => $getBucket['s3_bucket'],
            'Key' => $s3Key[1]
        ]);
        $request = $s3Client->createPresignedRequest($cmd, '+5 minutes');
        $presignedUrl = (string) $request->getUri();
        $event->setOffsiteUrl($presignedUrl);
    }

Reply:
The PatientRetrieveOffsiteDocument->url property is marked private so I'm not sure how you've been able to access it in your module.  That's why I exposed the getOpenEMRDocumentUrl() method.